lib/afmotion/http_client.rb

```diff
@@ -62,6 +65,12 @@ def request_serializer(serializer)
       else
         @operation_manager.requestSerializer = serializer
       end
+      reinit_options
+    end
+
+    def reinit_options
+      @headers.each{ |h,v| header(h, v) } if !@headers.nil? && !@headers.empty?
```

Comment:
Ahh you're right... definitely don't want to be looping through and possibly changing an object at the same time.